lib/backend/k8s/conversion/conversion_test.go

```diff
@@ -103,6 +103,23 @@ var _ = Describe("Test parsing strings", func() {
 	})
 })
 
+var _ = Describe("Test ServiceAccount name conversion", func() {
+	DescribeTable("service account conversion table",
+		func(name, expected string) {
+			Expect(SanitizeServiceAccountName(name)).To(Equal(expected))
+		},
+
+		Entry("should NOT change short names", "default", "default"),
+		Entry("should NOT change 63 character names", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk"),
+		Entry("should change 64 character names", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwx-93249d4c2f89"),
+
+		// The conversion code splits the input string at 50 characters, so test when a "-" falls around that area.
+		Entry("should change long names with a hyphen at the cutoff", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwx-yzabcdefghijkl", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwx-c086e39cde0f"),
+		Entry("should change long names with a hyphen before the cutoff", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvw-xyzabcdefghijkl", "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvw-cb2031a84741"),
```

Comment:
should check putting the output from one run as input into another and check it gives you a different result.